pkg/dbs/studydb/db_study-surveys.go

```diff
@@ -13,56 +15,132 @@ func (dbService *StudyDBService) SaveSurvey(instanceID string, studyKey string,
+
 }
 
-func (dbService *StudyDBService) FindSurveyDef(instanceID string, studyKey string, surveyKey string) (types.Survey, error) {
+func (dbService *StudyDBService) FindAllCurrentSurveyDefsForStudy(instanceID string, studyKey string, includeUnpublished bool, onlyInfos bool) (surveys []*types.Survey, err error) {
+	surveyKeys, err := dbService.GetSurveyKeysInStudy(instanceID, studyKey, includeUnpublished)
+	if err != nil {
+		return surveys, err
+	}
+	for _, key := range surveyKeys {
+		survey, err := dbService.FindCurentSurveyDef(instanceID, studyKey, key, onlyInfos)
+		if err != nil {
+			logger.Error.Println(err)
+			continue
+		}
+		surveys = append(surveys, survey)
+	}
+	return surveys, nil
+}
+
+func (dbService *StudyDBService) FindCurentSurveyDef(instanceID string, studyKey string, surveyKey string, onlyInfos bool) (surveys *types.Survey, err error) {
 	ctx, cancel := dbService.getContext()
 	defer cancel()
 
-	filter := bson.M{"current.surveyDefinition.key": surveyKey}
+	filter := bson.M{
+		"surveyDefinition.key": surveyKey,
+		"unpublished":          0,
+	}
 
-	elem := types.Survey{}
-	err := dbService.collectionRefStudySurveys(instanceID, studyKey).FindOne(ctx, filter).Decode(&elem)
+	elem := &types.Survey{}
+	opts := &options.FindOneOptions{
+		Sort: bson.D{
+			primitive.E{Key: "published", Value: -1},
+		},
+	}
+	if onlyInfos {
+		projection := bson.D{
+			primitive.E{Key: "surveyDefinition.items", Value: 0},
+			primitive.E{Key: "prefillRules", Value: 0},
+			primitive.E{Key: "contextRules", Value: 0},
+		}
+		opts.SetProjection(projection)
+	}
+
+	err = dbService.collectionRefStudySurveys(instanceID, studyKey).FindOne(ctx, filter, opts).Decode(&elem)
+	return elem, err
+}
+
+func (dbService *StudyDBService) FindSurveyDefByVersionID(instanceID string, studyKey string, surveyKey string, versionID string) (surveys *types.Survey, err error) {
+	ctx, cancel := dbService.getContext()
+	defer cancel()
+
+	filter := bson.M{
+		"surveyDefinition.key": surveyKey,
+		"versionID":            versionID,
+	}
+
+	elem := &types.Survey{}
+	err = dbService.collectionRefStudySurveys(instanceID, studyKey).FindOne(ctx, filter).Decode(&elem)
 	return elem, err
 }
 
-func (dbService *StudyDBService) FindAllSurveyDefsForStudy(instanceID string, studyKey string, onlyInfos bool) (surveys []types.Survey, err error) {
+func (dbService *StudyDBService) DeleteSurveyVersion(instanceID string, studyKey string, surveyKey string, versionID string) error {
+	ctx, cancel := dbService.getContext()
+	defer cancel()
+
+	filter := bson.M{
+		"surveyDefinition.key": surveyKey,
+		"versionID":            versionID,
+	}
+	res, err := dbService.collectionRefStudySurveys(instanceID, studyKey).DeleteOne(ctx, filter)
+	if res.DeletedCount < 1 {
+		err = errors.New("not found")
+	}
+	return err
+}
+
+func (dbService *StudyDBService) FindSurveyDefHistory(instanceID string, studyKey string, surveyKey string, onlyInfos bool) (surveys []*types.Survey, err error) {
```

Comment:
Good point. This method is used for:
 1) to retrieve it for researcher, and there the client can provide the appropriate sorting 
 2) data exporter which will attempt to lookup the corresponding versionId to match the survey definition.

For this 2) use-case sorting by the `published`  value could be useful as a fallback mechanism to find the correct version, if version IDs are not present / missing. 
Since once index already contains published, I assume it will be efficient already, without the need to add an additional index?

Reply:
I don't think it's relevant if sorted in ascending or descending order, so might just take it in descending, since the index has that already.

Reply:
unfortunately the index that we have wouldn't cover the queries that this method can generate. To fully cover both queries we need

`{ "published": 1, "surveyDefinition.key": 1 }`

Reply:
Added the sorting for this query using:
`bson.D{
		primitive.E{Key: "published", Value: -1},
	}
`

Also added the creation of the extra index - here, are you sure, we want to used "published": 1, and not -1? (You know mongoDB indexes certainly better, just double checking, since it easy to miss a dash)

Reply:
It's fine to put -1 in the definition but it doesn't really matter in this case, it only matters when sorting on more than one field

Reply:
I reopened it because we are still missing the sorting on published, right?

Reply:
Isn't that in line 196 covers that? I find it a little bit confusing, since some comments seem to be shifted / out of context :-)

Reply:
Yes I missed that, it's now resolved